### PR TITLE
Complete Use of the CacheManager

### PR DESCRIPTION
**Pull request by rubenporras**
Use the CacheManager in MonitoredClusteringBuilderState,
ImportedNamesTypesAdapter and AbstractResourceDescriptionDelta.

### CODE REVIEW

File: com.avaloq.tools.ddk.xtext/src/com/avaloq/tools/ddk/xtext/linking/ImportedNamesTypesAdapter.java [hunk 1]
```diff
@@ -31,7 +31,7 @@
  */
 public class ImportedNamesTypesAdapter extends ImportedNamesAdapter {
 
-  private final Map<QualifiedName, Set<EClass>> importedNamesTypes = Maps.newHashMap();
+  private final Map<QualifiedName, Set<EClass>> importedNamesTypes = CacheManager.getInstance().createMapCache("ImportedNamesTypesAdapter#importedNamesTypes"); //$NON-NLS-1$
```
Comment (mrubanov) [inline com.avaloq.tools.ddk.xtext/src/com/avaloq/tools/ddk/xtext/linking/ImportedNamesTypesAdapter.java]: this isn't really a cache either. We collect the names of objects imported in the given resource and then write it to the db...

File: com.avaloq.tools.ddk.xtext/src/com/avaloq/tools/ddk/xtext/resource/AbstractResourceDescriptionDelta.java [hunk 2]
```diff
@@ -38,7 +38,7 @@
   public <T> void addExtensionData(final Class<T> clazz, final T data) {
     if (data != null) {
       if (extensions == null) {
-        extensions = Maps.newHashMap();
+        extensions = CacheManager.getInstance().createMapCache("AbstractResourceDescriptionDelta#extensions"); //$NON-NLS-1$
```
[the remark this answers is not in the document]
Comment (rubenporras) [reply com.avaloq.tools.ddk.xtext/src/com/avaloq/tools/ddk/xtext/resource/AbstractResourceDescriptionDelta.java]: Thanks, I have removed it from the change.

Comment (mrubanov) [inline com.avaloq.tools.ddk.xtext/src/com/avaloq/tools/ddk/xtext/resource/AbstractResourceDescriptionDelta.java]: hm, i thought you would revert the change in this file...
